org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterThermostatRunningMode.java

```diff
@@ -0,0 +1,149 @@
+                cluster.setReporting(attribute, REPORTING_PERIOD_DEFAULT_MIN, REPORTING_PERIOD_DEFAULT_MAX, 0.1);
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            logger.error("{}: Exception setting reporting ", endpoint.getIeeeAddress(), e);
+        }
+
+        // Add a listener, then request the status
+        cluster.addAttributeListener(this);
+
+        return true;
+    }
+
+    @Override
+    public void disposeConverter() {
+        cluster.removeAttributeListener(this);
+    }
+
+    @Override
+    public void handleRefresh() {
+        cluster.getLocalTemperature(0);
+    }
+
+    @Override
+    public Channel getChannel(ThingUID thingUID, ZigBeeEndpoint endpoint) {
+        ZclThermostatCluster cluster = (ZclThermostatCluster) endpoint.getInputCluster(ZclThermostatCluster.CLUSTER_ID);
+        if (cluster == null) {
+            logger.trace("{}: Thermostat cluster not found", endpoint.getIeeeAddress());
+            return null;
+        }
+
+        try {
+            if (!cluster.discoverAttributes(false).get()) {
+                // Device is not supporting attribute reporting - instead, just read the attributes
+                Integer capabilities = cluster.getSystemMode(Long.MAX_VALUE);
+                if (capabilities == null) {
+                    logger.trace("{}: Thermostat running mode returned null", endpoint.getIeeeAddress());
+                    return null;
+                }
+            } else if (!cluster.isAttributeSupported(ZclThermostatCluster.ATTR_THERMOSTATRUNNINGMODE)) {
+                logger.trace("{}: Thermostat running mode not supported", endpoint.getIeeeAddress());
+                return null;
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            logger.warn("{}: Exception discovering attributes in thermostat cluster", endpoint.getIeeeAddress(), e);
+        }
+
+        // Sequence of operation defines the allowable modes
+        Integer states = cluster.getControlSequenceOfOperation(Long.MAX_VALUE);
+
+        // Define the allowable states
+        List<StateOption> options = new ArrayList<>();
+        options.add(new StateOption("0", "Off"));
+        options.add(new StateOption("1", "Auto"));
+        if (states != null && states != 0 && states != 1) {
+            options.add(new StateOption("4", "Heat"));
+            options.add(new StateOption("5", "Emergency Heating"));
+        }
+        if (states != null && states != 3 && states != 6) {
+            options.add(new StateOption("3", "Cool"));
+            options.add(new StateOption("6", "Precooling"));
+        }
+        options.add(new StateOption("7", "Fan Only"));
+        options.add(new StateOption("8", "Dry"));
+        options.add(new StateOption("9", "Sleep"));
+
+        StateDescription description = new StateDescription(BigDecimal.ZERO, BigDecimal.valueOf(9),
```

Comment:
The `StateDescription` is currently not used - should it be added to the channel definition?

Reply:
Probably, yes. However I don't see a method to do this here which might be why I didn't do it ;) Do you know how this can be set here rather than in the XML?

Reply:
The only option I could find is implementing an `org.eclipse.smarthome.core.thing.type.DynamicStateDescriptionProvider`. (This is somewhat similar to what I added with the `DynamicWarningCommandDescriptionProvider ` in #372, compare https://github.com/openhab/org.openhab.binding.zigbee/pull/372/files#diff-5c98c205a30a6ec68538422ebd96ab47).